Formula/openvino.rb

```diff
@@ -0,0 +1,188 @@
+      sha256 "245e19b4a2c926aedb764fecbd02c5023deab56eba497e4a3c8194e760e26413"
+    end
+
+    resource "arm_compute" do
+      url "https://github.com/ARM-software/ComputeLibrary/archive/refs/tags/v22.11.tar.gz"
+      sha256 "e20a060d3c4f803889d96c2f0b865004ba3ef4e228299a44339ea1c1ba827c85"
+    end
+  end
+
+  resource "onednn_cpu" do
+    on_intel do
+      url "https://github.com/openvinotoolkit/oneDNN/archive/44de3c3698b687c26e487fc8f0213fa487e8fe2c.tar.gz"
+      sha256 "2c6aa7d787a947aa032224683f216ab38c168de4aed61d0554671774060a3615"
+    end
+  end
+
+  resource "ade" do
+    url "https://github.com/opencv/ade/archive/refs/tags/v0.1.1f.tar.gz"
+    sha256 "c316680efbb5dd3ac4e10bb8cea345cf26a6a25ebc22418f8f0b8ca931a550e9"
+  end
+
+  resource "ittapi" do
+    url "https://github.com/intel/ittapi/archive/refs/tags/v3.23.0.tar.gz"
+    sha256 "9af1231808c602c2f7a66924c8798b1741d3aa4b15f3874d82ca7a89b5dbb1b1"
+  end
+
+  resource "xbyak" do
+    url "https://github.com/herumi/xbyak/archive/refs/tags/v6.63.tar.gz"
+    sha256 "16c60f0682502624115c4dc9fec66782ae68ef32e469946f50cd169179ea92bb"
+  end
+
+  resource "onnx" do
+    url "https://github.com/onnx/onnx/archive/refs/tags/v1.12.0.tar.gz"
+    sha256 "052ad3d5dad358a33606e0fc89483f8150bb0655c99b12a43aa58b5b7f0cc507"
+  end
+
+  def install
+    resource("ade").stage buildpath/"thirdparty/ade"
+    resource("ittapi").stage buildpath/"thirdparty/ittapi/ittapi"
+    resource("xbyak").stage buildpath/"thirdparty/xbyak"
+    resource("onnx").stage buildpath/"thirdparty/onnx/onnx"
+
+    if Hardware::CPU.arm?
+      resource("openvino_contrib").stage buildpath/"openvino_contrib"
+      resource("arm_compute").stage buildpath/"openvino_contrib/modules/arm_plugin/thirdparty/ComputeLibrary"
+    else
+      resource("onednn_cpu").stage buildpath/"src/plugins/intel_cpu/thirdparty/onednn"
+    end
+
+    cmake_args = std_cmake_args + %w[
+      -DCMAKE_OSX_DEPLOYMENT_TARGET=
+      -DENABLE_TBBBIND_2_5=OFF
+      -DENABLE_CPPLINT=OFF
+      -DENABLE_CLANG_FORMAT=OFF
+      -DENABLE_NCC_STYLE=OFF
+      -DENABLE_INTEL_GPU=OFF
+      -DENABLE_TEMPLATE=OFF
+      -DENABLE_INTEL_GNA=OFF
+      -DENABLE_INTEL_MYRIAD_COMMON=OFF
+      -DENABLE_PYTHON=OFF
+      -DENABLE_SAMPLES=OFF
+      -DENABLE_COMPILE_TOOL=OFF
+      -DCPACK_GENERATOR=BREW
+      -DENABLE_SYSTEM_PUGIXML=ON
+      -DENABLE_SYSTEM_TBB=ON
+      -DENABLE_SYSTEM_PROTOBUF=ON
+    ]
+
+    if Hardware::CPU.arm?
+      cmake_args += %W[
+        -DOPENVINO_EXTRA_MODULES=#{buildpath}/openvino_contrib/modules/arm_plugin
+      ]
+    end
+
+    system "cmake", "-S", buildpath.to_s, "-B", "#{buildpath}/openvino_build", *cmake_args
+    system "cmake", "--build", "#{buildpath}/openvino_build"
+
+    # install only required components
+
+    components = %w[core core_dev
+                    cpu gpu batch multi hetero
+                    ir onnx paddle pytorch tensorflow]
```

Comment:
Won't these last 2 conflict with the formula?

Reply:
no, these components are responsible for reading ir, onnx, paddle, pytorch andtensorflow files formats accordingly.  They are totally independent on actual TF or PyTorch.